kafka-connect-s3/docs/configuration_options.rst

```diff
@@ -109,6 +109,14 @@ S3
   * Valid Values: Any class implementing: interface com.amazonaws.auth.AWSCredentialsProvider
   * Importance: low
 
+``s3.credentials.provider.configs.``
+  The properties that begin with this prefix will be used to configure a class, specified by
+  ``s3.credentials.provider.class`` if it implements org.apache.kafka.common.Configurable.
+
+  * Type: set<object>
+  * Importance: low
+
+
```

Comment:
Thanks, I haven't seen `main()` method initially. Does it make sense to add autogeneration of this file to `pom.xml` in this PR as well?
Since this file's content is generated by `main()` - I think it's better to add custom credentials section to https://github.com/confluentinc/kafka-connect-storage-cloud/blob/master/kafka-connect-s3/docs/s3_connector.rst

Reply:
I agree with @kkonstantine. That would be nice to do for all of our connectors, but to do this we probably also need to generate the remainder of the files. We have one connector that does this autogeneration (as a test, currently), and in fact a fair amount of the documentation is generated from code so that it can reuse the constants and other things defined in the code.